src/main/java/co/cask/PluginJSON.java

```diff
@@ -244,23 +319,39 @@ public void execute()  throws MojoExecutionException {
   /**
    * Initializes this Mojo, extracts all necessary paths and configurations.
    */
-  private void initialize() {
+  private void initialize() throws MojoExecutionException {
     groupId = project.getGroupId();
     artifactId = project.getArtifactId();
     version = project.getVersion();
     baseDirectory = project.getBasedir();
     buildDirectory = new File(project.getBuild().getDirectory());
-    if (!widgetsDirectory.contains("/")) {
-      widgetDirectory = new File(project.getBasedir() + "/" + widgetsDirectory);
+    widgetDirectory = getAndValidate(baseDirectory, widgetsDirectory);
+    iconDirectory = getAndValidate(baseDirectory, iconsDirectory);
+    docDirectory = getAndValidate(baseDirectory, docsDirectory);
+  }
+
+  /**
+   * Create directories based on given project
+   */
+  private File getAndValidate(File baseDirectory, String newDirectoryName) throws MojoExecutionException {
```

Comment:
don't call this newDirectoryName, you're not creating a new directory. Call it directoryPath.